cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1288,9 +1295,12 @@ func generateMountsCommonInstallRecover(mst *initramfsMountsState) (model *asser
 	//            mounted, we should also implement writable-paths here too as
 	//            writing it in Go instead of shellscript is desirable
 
-	// 2.3. mount "ubuntu-data" on a tmpfs
+	// 2.3. mount "ubuntu-data" on a tmpfs, and also mount with nosuid to prevent
+	// snaps from being able to bypass the sandbox by creating suid root files
+	// there and escape the sandbox
```

Comment:
s/escape/trying to escape/

AFAIU even though the process would then be running as root there's still seccomp and apparmor that should prevent further escalation, so 'trying to escape' (or circumvent like in the earlier comment)?

Reply:
Does snapd install apparmor profiles with an attachment specification that covers all the writable areas available to each snap to write to?

Reply:
Snaps when they execute always transition to their apparmor profile for the app in the snap that is being executed, but we do not have attachment specifications to cover all the writable areas that snaps can write to.